python/ray/_private/runtime_env/packaging.py

```diff
@@ -393,34 +434,140 @@ def download_and_unpack_package(
                     raise IOError(f"Failed to fetch URI {pkg_uri} from GCS.")
                 code = code or b""
                 pkg_file.write_bytes(code)
-            elif protocol == Protocol.S3:
-                # Download package from S3.
-                try:
-                    from smart_open import open
-                    import boto3
-                except ImportError:
-                    raise ImportError(
-                        "You must `pip install smart_open` and "
-                        "`pip install boto3` to fetch URIs in s3 "
-                        "bucket.")
-
-                tp = {"client": boto3.client("s3")}
+                unzip_package(
+                    package_path=pkg_file,
+                    target_dir=local_dir,
+                    remove_top_level_directory=False,
+                    unlink_zip=True,
+                    logger=logger)
+            elif protocol in Protocol.remote_protocols():
+                # Download package from remote URI
+                tp = None
+
+                if protocol == Protocol.S3:
+                    try:
+                        from smart_open import open
+                        import boto3
+                    except ImportError:
+                        raise ImportError(
+                            "You must `pip install smart_open` and "
+                            "`pip install boto3` to fetch URIs in s3 "
+                            "bucket.")
+                    tp = {"client": boto3.client("s3")}
```

Comment:
hmm do we not need something similar for GS here?

Reply:
Users can pass a GS client into ``smart_open``'s ``open`` function, and ``smart_open`` will use that client's credentials when accessing the GS. If you don't pass anything in, ``smart_open`` [will use the user's existing ``google-cloud`` package to authenticate](https://pypi.org/project/smart-open/).

Since we don't take in the user's credentials for ``working_dir`` or ``py_modules``, there's no reason to manually initialize a client for GS.

As a side note, I noticed that S3 seems to work even without the ``tp`` definition. If we don't include that parameter, ``smart_open`` generates its own ``boto3`` client under the hood. The ``smart_open`` [documentation](https://github.com/RaRe-Technologies/smart_open/blob/develop/howto.md) suggests that defining ``tp`` once and then passing it in to successive ``open`` calls can improve efficiency, but our code doesn't make successive ``open`` calls. I'm hesitant to remove the line because I haven't dug deep enough to confidently say it does nothing, but I think it's something to keep in mind.

Reply:
I see, gotcha. The main thing I'm concerned about is that we raise a good error message when the `google-cloud` package isn't installed. Can we do a similar import check and throw that? Otherwise LGTM